### ISSUE TITLE
Vim `TRUE` is not Lua `true`

### ISSUE THREAD

**Issue by otommod** [neovim/neovim#15045]
- `nvim --version`: v0.5.0

Like the title says, VimL functions that return `TRUE` and `FALSE` are turned into numbers in Lua.  That would be fine in other languages but in this case, Lua treats `0` as true which makes it tedious to use said functions and also quite surprising and error-prone in case you forget.

At this point, I don't think this is fixable, there's a lot of code that checks against `1` and `0` that'd just instantly break.  However, I think there should at least be some warning in the docs.

**Comment by clason** [comment]
Yes, that is a known problem and documented in the (semi-)inofficial guide: https://github.com/nanotee/nvim-lua-guide#caveats-3

The long term goal is to expose vim internals directly to Lua without needing to go through vimscript, making this moot.
